torchtext/datasets/text_classification.py

```diff
@@ -99,6 +107,9 @@ def get_vocab(self):
 
 
 def _setup_datasets(dataset_name, root='.data', ngrams=2, vocab=None):
+    if not os.path.exists(root):
```

Comment:
I have an error that is thrown if the user doesn't do this. I don't think we should create a directory (and in particular many directories) if it doesn't exist. For example, a user might supply a very long, wrong directory by accident.

Reply:
Sure. If the error message is desired, we don't need to create a directory here. I will remove it.